### PR TITLE
fix: Normalize the dotnetcore service name so that it conforms to the Kubernetes standard

### PR DESCRIPTION
**Pull request by Akash-Nayak**
Signed-off-by: Akash Nayak <akash19nayak@gmail.com>

### CODE REVIEW

File: transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go [hunk 1]
```diff
@@ -386,9 +386,9 @@ func (t *DotNetCoreDockerfileGenerator) Transform(newArtifacts []transformertype
 					logrus.Errorf("Unable to read the launchSettings.json file: %s", err)
 					continue
 				}
-				if _, ok := launchSettings.Profiles[projectName]; ok {
-					if launchSettings.Profiles[projectName].ApplicationURL != "" {
-						Urls := strings.Split(launchSettings.Profiles[projectName].ApplicationURL, ";")
+				if _, ok := launchSettings.Profiles[dotNetCoreTemplateConfig.AppName]; ok {
+					if launchSettings.Profiles[dotNetCoreTemplateConfig.AppName].ApplicationURL != "" {
+						Urls := strings.Split(launchSettings.Profiles[dotNetCoreTemplateConfig.AppName].ApplicationURL, ";")
```
Comment (ashokponkumar) [inline transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go:391]: Create a new variable say, dotnetServiceName to access here. Use template config only for templates.

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Thanks @Akash-Nayak . Minor suggestions.

File: transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go [hunk 2]
```diff
@@ -374,7 +374,7 @@ func (t *DotNetCoreDockerfileGenerator) Transform(newArtifacts []transformertype
 		publishProfileFilesPath := findPublishProfiles(filepath.Join(a.Paths[artifacts.ProjectPathPathType][0], csprojRelFilePath))
 		dotNetCoreTemplateConfig.PublishProfileFilePath, dotNetCoreTemplateConfig.PublishUrl = getPublishProfile(publishProfileFilesPath, a.Name, a.Paths[artifacts.ProjectPathPathType][0])
 		dotNetCoreTemplateConfig.CsprojFilePath = csprojRelFilePath
-		projectName := strings.TrimSuffix(filepath.Base(dotNetCoreTemplateConfig.CsprojFilePath), filepath.Ext(dotNetCoreTemplateConfig.CsprojFilePath))
+		dotNetCoreTemplateConfig.AppName = strings.TrimSuffix(filepath.Base(dotNetCoreTemplateConfig.CsprojFilePath), filepath.Ext(dotNetCoreTemplateConfig.CsprojFilePath))
```
[the remark this answers is not in the document]
Comment (Akash-Nayak) [reply transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go]: I have changed it to `CsprojFileName` in the template.

**Review by ashokponkumar — APPROVED** [review]

File: transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go (hunk 1 above)
Comment (Akash-Nayak) [reply transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go]: As the dotnet service name (`appName`) is already decided in the detect phase, so I have named the variable `dotnetProjectName`.

**Pull request merged**